server/lib/processLogs.js

```diff
@@ -0,0 +1,143 @@
+const async = require('async');
+const moment = require('moment');
+const Request = require('request');
+const loggingTools = require('auth0-log-extension-tools');
+
+const config = require('./config');
+const logger = require('./logger');
+
+module.exports = storage =>
+  (req, res, next) => {
+    const wtBody = (req.webtaskContext && req.webtaskContext.body) || req.body || {};
+    const wtHead = (req.webtaskContext && req.webtaskContext.headers) || {};
+    const isCron = (wtBody.schedule && wtBody.state === 'active') || (wtHead.referer === 'https://manage.auth0.com/' && wtHead['if-none-match']);
+
+    if (!isCron) {
+      return next();
+    }
+
+    const url = config('WEBHOOK_URL');
+    const batchMode = config('SEND_AS_BATCH') === true || config('SEND_AS_BATCH') === 'true';
+    const concurrentCalls = parseInt(config('WEBHOOK_CONCURRENT_CALLS'), 10) || 5;
+
+    const sendRequest = (data, callback) =>
+      Request({
+        method: 'POST',
+        url: url,
+        json: true,
+        body: data
+      }, (err, response, body) => {
+        if (err || response.statusCode < 200 || response.statusCode >= 400) {
+          return callback(err || body || response.statusCode);
+        }
+
+        return callback();
+      });
+
+    const callWebhook = (logs, callback) => {
+      if (batchMode) {
+        return sendRequest(logs, callback);
+      }
+
+      return async.eachLimit(logs, concurrentCalls, sendRequest, callback);
+    };
+
+    const onLogsReceived = (logs, callback) => {
+      if (!logs || !logs.length) {
+        return callback();
+      }
+
+      let endpointsFilter = config('AUTH0_API_ENDPOINTS').split(',');
+      endpointsFilter = endpointsFilter.length > 0 && endpointsFilter[0] === '' ? [] : endpointsFilter;
+
+      const requestMatchesFilter = (log) => {
+        if (!endpointsFilter || !endpointsFilter.length) return true;
+        return log.details.request && log.details.request.path &&
```

Comment:
Let's split this up into multiple lines/if statements for readability. This one liner has a lot going on in it.